tests/scripts/all.sh

```diff
@@ -896,12 +896,29 @@ cd "$MBEDTLS_ROOT_DIR"
 rm -rf "$OUT_OF_SOURCE_DIR"
 unset MBEDTLS_ROOT_DIR
 
+# Test that the function mbedtls_platform_zeroize() is not optimized away by
+# different combinations of compilers and optimization flags by using an
+# auxiliary GDB script. Unfortunately, GDB does not return error values to the
+# system in all cases that the script fails, so we must manually search the
+# output to check whether the pass string is present and no failure strings
+# were printed.
 for optimization_flag in -O2 -O3 -Ofast -Os; do
     for compiler in clang gcc; do
         msg "test: $compiler $optimization_flag, mbedtls_platform_zeroize()"
         cleanup
-        CC="$compiler" DEBUG=1 CFLAGS="$optimization_flag" make programs
-        gdb -x tests/scripts/test_zeroize.gdb -nw -batch -nx
+        make programs CC="$compiler" DEBUG=1 CFLAGS="$optimization_flag"
+        if_build_succeeded gdb -x tests/scripts/test_zeroize.gdb -nw -batch -nx > test_zeroize.log 2>&1
+        if [ ! -s test_zeroize.log ]; then
+            err_msg "test_zeroize.log was not found or is empty"
+            record_status [ -s test_zeroize.log ]
```

Comment:
I think we want `if_build_succeeded` instead of `record_status` here.

Also, I'm not sure I agree with the general "if, err_msg, if_build_succeed" structure for reasons that include (1) error reporting and (2) what happens in the build failed? Why not go directly with
```
if_build_succeeded test -s test_zeroize.log
if_build_succeeded grep "The buffer was correctly zeroized" test_zeroize.log
if_build_succeeded not grep -i "error" test_zeroize.log
```
This is compact and solves the two issues mentioned.

(Technical point: we would have to define the obvious "not" function, which I'm already doing other PRs, see https://github.com/ARMmbed/mbedtls/pull/1617/commits/2adb375c50e2db5f44dd1ce8b7cb4b33b035563a#diff-d36612fab4653d3f01f967a3c1bfc170 for example.)

Reply:
Thanks for letting me know. I was very unsure what was the correct way of doing it, so I just followed the example of the out of tree build using Cmake (just above this code) that does something very similar to this. I have now fixed the code work as you suggested.